bin/install

```diff
@@ -43,9 +43,14 @@ install() {
 
 	local -r platform="$(_get_platform)"
 	local arch; arch="$(_get_arch)"
-	# Currently there is no darwin/arm support: https://github.com/koalaman/shellcheck/releases
-	# If/when it is added, remove the folling line and make arch a read-only variable.
-	test "$platform" == "darwin" && arch="x86_64"
+	# Shellcheck supports Darwin/ARM only starting from v0.10.0.
+	# If the requested version is lower than v0.10.0, then fall back to the x86_64 version.
+	local -r version_parts=(${version//./ })
+	if [ "${version_parts[0]}" -eq 0 ] && [ "${version_parts[1]}" -lt 10 ]; then
+		test "$platform" == "darwin" && arch="x86_64"
+	else
+		test "$platform" == "darwin" && arch="aarch64"
+	fi
```

Comment:
Is this meaning v0.10.0+ forced to replace x86_64 with aarch64?

Looks like macos-13 runner(x86-64) is failing since this change

- before: https://github.com/mise-plugins/asdf-yamlfmt/actions/runs/10413650234/job/28841366809

```
 Downloading shellcheck from https://github.com/koalaman/shellcheck/releases/download/v0.10.0/shellcheck-v0.10.0.darwin.x86_64.tar.xz to /Users/runner/.asdf/installs/shellcheck/0.10.0/bin
```

- after: https://github.com/mise-plugins/asdf-yamlfmt/actions/runs/10428790086/job/28885389050

```
 Downloading shellcheck from https://github.com/koalaman/shellcheck/releases/download/v0.10.0/shellcheck-v0.10.0.darwin.aarch64.tar.xz to /Users/runner/.asdf/installs/shellcheck/0.10.0/bin
+ shellcheck --shell=bash --external-sources bin/download bin/help.deps bin/help.links bin/help.overview bin/install bin/list-all --source-path=lib/ lib/utils.bash scripts/format.bash scripts/lint.bash
/nix/store/jv3hs0ryn3lihjw5xbjh2il74rfaynl4-asdf-vm-0.13.1/share/asdf-vm/lib/commands/command-exec.bash: line 28: /Users/runner/.asdf/installs/shellcheck/0.10.0/bin/shellcheck: Bad CPU type in executable
/nix/store/jv3hs0ryn3lihjw5xbjh2il74rfaynl4-asdf-vm-0.13.1/share/asdf-vm/lib/commands/command-exec.bash: line 28: /Users/runner/.asdf/installs/shellcheck/0.10.0/bin/shellcheck: Undefined error: 0
```

Reply:
I've sent a PR for this https://github.com/luizm/asdf-shellcheck/pull/12

Reply:
>Is this meaning v0.10.0+ forced to replace x86_64 with aarch64?

Yeah, pretty much. Depending on the requested version, this plugin downloads the binary for the current platform. Thanks for fixing that!

Reply:
thank you guys